pyrobosim/pyrobosim/planning/pddlstream/primitives.py

```diff
@@ -164,5 +236,5 @@ def test_collision_free(o1, p1, o2, p2, padding=0.0):
     :rtype: bool
     """
     o1_poly = transform_polygon(inflate_polygon(o1.raw_polygon, padding), p1)
-    o2_poly = transform_polygon(o2.raw_polygon, p2)
+    o2_poly = transform_polygon(inflate_polygon(o2.raw_polygon, padding), p2)
```

Comment:
Is this a case where you can use:

`o2_poly = o2.collision_polygon`

Reply:
It's not in the case of streams because `obj.polygon` and `obj.collision_polygon` exist at the actual pose of the object, whereas `obj.raw_polygon` is zero centered and used for hypothesizing about the object polygon at other poses during planning.